runtime/compiler/optimizer/J9ValuePropagation.cpp

```diff
@@ -2420,6 +2421,42 @@ J9::ValuePropagation::constrainRecognizedMethod(TR::Node *node)
                }
             break;
             }
+         case TR::java_lang_invoke_MethodHandle_invokeBasic:
+            {
+            TR::Node* mh = node->getFirstArgument();
+            bool isGlobal;
+            TR::VPConstraint* mhConstraint = getConstraint(mh, isGlobal);
+            if (mhConstraint
+               && mhConstraint->getKnownObject()
+               && mhConstraint->isNonNullObject())
+               {
+               if (!J9::TransformUtil::refineMethodHandleInvokeBasic(comp(), _curTree, node, mhConstraint->getKnownObject()->getIndex(), trace()))
+                  return;
+
+               TR_PrexArgInfo *argInfo = new (trStackMemory()) TR_PrexArgInfo(node->getNumChildren(), trMemory());
+               for (int32_t i = 0; i < node->getNumChildren(); i++)
+                  {
+                  TR::Node *argNode = node->getChild(i);
+                  if (argNode && argNode->getDataType() == TR::Address)
+                     {
+                     TR::VPConstraint *argConstraint = getConstraint(argNode, isGlobal);
+                     if (argConstraint
+                         && argConstraint->getKnownObject()
+                         && argConstraint->isNonNullObject())
+                        {
+                        argInfo->set(i, new (trStackMemory()) TR_PrexArgument(argConstraint->getKnownObject()->getIndex(), comp()));
+                        if (trace())
+                           traceMsg(comp(), "PREX.vp:    Child %d [%p] arg is known object obj%d\n", i, argInfo->get(i), argConstraint->getKnownObject()->getIndex());
+                        }
+                     }
+                  }
+               if (trace())
+                  traceMsg(comp(), "PREX.vp: Done populating prex argInfo for %s %p\n", node->getOpCode().getName(), node);
+
+               _refinedMethodHandleINLMethodsToInline.add(new (trStackMemory()) OMR::ValuePropagation::CallInfo(this, mhConstraint->getClass(), argInfo));
```

Comment:
At first I thought that if the `invokeBasic()` call were in a loop, this might add two entries to the list. But of course, that can't happen because if the call is refined the first time through, then when we reach the same call the second time through, it won't be calling `invokeBasic()` anymore, so we won't get here. Do you think that's worth a comment?